src/js/media-chrome-range.js

```diff
@@ -343,27 +381,79 @@ class MediaChromeRange extends globalThis.HTMLElement {
 
     this.shadowRoot.removeEventListener('focusin', this.#onFocusIn);
     this.shadowRoot.removeEventListener('focusout', this.#onFocusOut);
+    unobserveResize(this.container, this.#updateComputedStyles);
+  }
+
+  #updateComputedStyles = () => {
+    // This fixes a Chrome bug where it doesn't refresh the clip-path on content resize.
+    const clipping = this.shadowRoot.querySelector('#segments-clipping');
+    clipping.parentNode.append(clipping);
   }
 
   updatePointerBar(evt) {
-    // Get mouse position percent
-    const rangeRect = this.range.getBoundingClientRect();
-    let mousePercent = (evt.clientX - rangeRect.left) / rangeRect.width;
-    // Lock between 0 and 1
-    mousePercent = Math.max(0, Math.min(1, mousePercent)) * 100;
-
-    const { style } = getOrInsertCSSRule(this.shadowRoot, '#pointer');
-    style.setProperty('width', `${mousePercent}%`);
+    this.#cssRules.pointer.style.setProperty('width', `${this.#getPointerRatio(evt) * 100}%`);
   }
 
   updateBar() {
     const rangePercent = this.range.valueAsNumber * 100;
+    this.#cssRules.progress.style.setProperty('width', `${rangePercent}%`);
+    this.#cssRules.thumb.style.setProperty('left', `${rangePercent}%`);
+  }
+
+  updateSegments(segments) {
+    if (!segments?.length) return;
+
+    this.container.classList.toggle('segments', !!segments.length);
+
+    const clipping = this.shadowRoot.querySelector('#segments-clipping');
+    clipping.textContent = '';
+
+    const normalized = [...new Set([
+      +this.range.min,
+      ...segments.flatMap(s => [s.start, s.end]),
+      +this.range.max
+    ])];
+
+    this.#segments = [...normalized];
 
-    const progressRule = getOrInsertCSSRule(this.shadowRoot, '#progress');
-    const thumbRule = getOrInsertCSSRule(this.shadowRoot, '#thumb');
+    const lastMarker = normalized.pop();
+    for (const [i, marker] of normalized.entries()) {
+      const [isFirst, isLast] = [i === 0, i === normalized.length - 1];
+      const x = isFirst ? 'calc(var(--segments-gap) / -1)' : `${marker * 100}%`;
+      const x2 = isLast ? lastMarker : normalized[i + 1];
+      const width = `calc(${(x2 - marker) * 100}%${isFirst || isLast ? '' : ` - var(--segments-gap)`})`;
 
-    progressRule.style.setProperty('width', `${rangePercent}%`);
-    thumbRule.style.setProperty('left', `${rangePercent}%`);
+      const segmentEl = document.createElementNS('http://www.w3.org/2000/svg', 'rect');
+      const cssRule = getOrInsertCSSRule(this.shadowRoot, `#segments-clipping rect:nth-child(${i + 1})`);
+      cssRule.style.setProperty('x', x);
+      cssRule.style.setProperty('width', width);
+      clipping.append(segmentEl);
+    }
```

Comment:
fyi: I didn't write this in one go 😅 
was reworked many times over. 
turns out, at least for me, positioning and sizing unequal segments with a constant gap is not as easy as it seems.
there is also an offset applied to the whole `#segments-clipping` in the CSS above